apps/web/src/components/importer/components/file-provider-handler.tsx

```diff
@@ -0,0 +1,324 @@
+/*
+This file is part of the Notesnook project (https://notesnook.com/)
+
+Copyright (C) 2023 Streetwriters (Private) Limited
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import {
+  IFile,
+  IFileProvider,
+  Note,
+  ProviderSettings,
+  transform
+} from "@notesnook-importer/core";
+import { formatBytes } from "@notesnook/common";
+import { ScrollContainer } from "@notesnook/ui";
+import { Button, Flex, Input, Text } from "@theme-ui/components";
+import { xxhash64 } from "hash-wasm";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useDropzone } from "react-dropzone";
+import { importNote } from "../../../utils/importer";
+import Accordion from "../../accordion";
+import { TransformResult } from "../types";
+
+type FileProviderHandlerProps = {
+  provider: IFileProvider;
+  onTransformFinished: (result: TransformResult) => void;
+};
+
+type Progress = {
+  total: number;
+  done: number;
+};
+
+export function FileProviderHandler(props: FileProviderHandlerProps) {
+  const { provider, onTransformFinished } = props;
+  const [files, setFiles] = useState<File[]>([]);
+  const [filesProgress, setFilesProgress] = useState<Progress>({
+    done: 0,
+    total: 0
+  });
+  const [totalNoteCount, setTotalNoteCount] = useState(0);
+  const [_, setCounter] = useState<number>(0);
+  const logs = useRef<string[]>([]);
+
+  const onDrop = useCallback((acceptedFiles: File[]) => {
+    setFiles((files) => {
+      const newFiles = [...acceptedFiles, ...files];
+      return newFiles;
+    });
+  }, []);
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    accept: {
+      file: provider?.supportedExtensions?.concat([".zip"])
+    }
+  });
+
+  useEffect(() => {
+    setFiles([]);
+  }, [provider]);
+
+  async function onStartImport() {
+    let processedNotes: Note[] = [];
```

Comment:
No need to store the processed notes in memory. That's exactly what we want to avoid so users can import insanely huge amount of notes.